content/300-guides/050-database/100-developing-with-prisma-migrate/180-generating-down-migrations.mdx

```diff
@@ -0,0 +1,122 @@
+---
+title: Generating down migrations
+metaTitle: Generating down migrations
+metaDescription: How to generate down migrations
+tocDepth: 2
+---
+
+<TopBlock>
+
+This guide describes how to generate a down migration that reverses a given [migration file](/concepts/components/prisma-migrate#migration-history).
+
+</TopBlock>
+
+## About down migrations
+
+!!intro text
+
+## Considerations when generating down migrations
+
+!!text here:
+
+- manual SQL lost?
```

Comment:
Yes, `migrate diff` will only be able to diff what it understands.
So custom SQL that is not understood by Migrate, or SQL that are not touching the database schema (like data migration) will not make it into the diff

Reply:
`migrate diff` compares the database schema from one state to another and creates "a diff" to get to it

Reply:
Kind of, if you use manual SQL, you need to do this:

- create the down migration
- create the up migrate with --create-only so it can be edited before its applied to the DB
- manually add your up SQL to up migration (e.g. create table)
- manually add the inverted SQL to the down migration (e.g. drop table)

Reply:
Note:
I'm not sure `(e.g. create table)` and `(e.g. drop table)` are good examples here, since this is something supported by Migrate.
It might be confusing to use them, as is, as a custom SQL example?

Reply:
Thanks, added a brief explanation.

Reply:
Thanks, added this